approval_polls/views.py

```diff
@@ -1,101 +1,212 @@
+            ballot.save()
 
-  for counter, choice in enumerate(poll.choice_set.all()):
-    if 'choice' + str(counter + 1) in request.POST:
-      ballot.vote_set.create(choice = choice)
-      ballot.save()
+    poll.save()
 
-  poll.save()
+    return HttpResponseRedirect(reverse('approval_polls:results', args=(poll.id,)))
 
-  return HttpResponseRedirect(reverse('approval_polls:results', args=(poll.id,)))
 
 def embed_instructions(request, poll_id):
-  link = request.build_absolute_uri('/approval_polls/{}'.format(poll_id))
-  return render(request, 'approval_polls/embed_instructions.html', {'link': link})
+    link = request.build_absolute_uri('/approval_polls/{}'.format(poll_id))
+    return render(request, 'approval_polls/embed_instructions.html', {'link': link})
 
-class CreateView(generic.View):
 
-  @method_decorator(login_required)
-  def get(self, request, *args, **kwargs):
-    return render(request, 'approval_polls/create.html')
+class PollUpdate(generic.View):
+    '''A generic update view for our poll.
+
+    Since this is actually a compound form updating two different models
+    a Poll and "n" number of Choice instances we cannot use the straight forward
+    UpdateView.
+    '''
+
+
+    success_url = '/approval_polls/my-polls'
+
+    @method_decorator(login_required)
+    def get(self, request, pk):
+        '''We have to get the object and the choices.
+
+        Arguments:
+            request (WSGIRequest): the standard request that django calls the get method with
+            pk (int): the id of the poll object we are mean to edit
+
+        Returns:
+            HTTPResponse
+        '''
+
+        form = PollUpdateForm(instance=Poll.objects.get(pk=pk))
+
+        choices = [{'form': ChoiceForm(instance=choice), 'id': choice.id} for choice in Choice.objects.filter(poll__pk=pk)]
+        my_context = {'form': form, 'choices': choices}
+        request_context = RequestContext(request)
+
+        return render_to_response('approval_polls/poll_update_form.html',
+                                  my_context, context_instance=request_context)
+
+    @method_decorator(login_required)
+    def post(self, request, pk):
+        '''Recieving the updated data for the Poll and it's choices
+
+        Arguments:
+            request (WSGIRequest): The request.POST member will contain all the data
+            from the form. This must be separated out and validated manually.
+
+            pk (int): The primary key of the Poll being updated.
+
+        Returns:
+            WSGIResponse: Upon success it redirects to "my-polls". Upon any failure
+            it returns to the poll editing page with the information pre-filled and any
+            error messages.
+        '''
+
+        # First get the data from the post and sort it out.
+        for k in ['question', 'open_date', 'close_date', 'suspended']:
+            try:
+                poll_data[k] = request.POST.get(k)
+            except NameError:
```

Comment:
Well I get what you are saying there are a couple of reasons for this. 

1) In python variables are not declared. Unlike strictly typed languages there is no declaration for a variable in python. The paradigm of declaration is a hold over from strictly typed languages. 

2) "try" is highly optimized in python. Unlike other languages. Java, perl, C, C++ where try is very expensive in Python it is rather trivial. So trivial in fact that some tome around 2005 or maybe earlier I'm kind of fuzzy on the date Alex Martelli floated the idea of changing 
try:
<some code>
except <exception>

Into
try:
<some code>
experience <exception || condition:
<some code>

So while we can go ahead and declare like a strictly typed language it is less pythonic to do so.